settlement-pipelines/src/reporting.rs

```diff
@@ -45,45 +50,267 @@ impl<T: PrintReportable> DerefMut for ReportHandler<T> {
+    }
+
+    pub fn error(message: impl Into<String>, vote_account: impl Into<Pubkey>) -> Self {
+        Self::new(message, ErrorSeverity::Error, vote_account)
+    }
+
+    pub fn retryable(message: impl Into<String>, vote_account: impl Into<Pubkey>) -> Self {
+        Self::new(message, ErrorSeverity::RetryableError, vote_account)
+    }
+
+    // Delegate common methods to base
+    pub fn severity(&self) -> ErrorSeverity {
+        self.base.severity
+    }
+
+    pub fn message(&self) -> &str {
+        &self.base.message
+    }
+
+    pub fn source(&self) -> Option<&str> {
+        self.base.source.as_deref()
+    }
+}
+
+impl Display for VoteAccountError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{} (vote_account: {})", self.base, self.vote_account)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum ErrorEntry {
+    Generic(GenericError),
+    VoteAccount(VoteAccountError),
+}
+
+impl ErrorEntry {
+    pub fn severity(&self) -> ErrorSeverity {
+        match self {
+            ErrorEntry::Generic(err) => err.severity,
+            ErrorEntry::VoteAccount(err) => err.severity(),
+        }
+    }
+
+    pub fn is_retryable(&self) -> bool {
+        self.severity().is_retryable()
+    }
+
+    pub fn is_critical(&self) -> bool {
+        self.severity().is_critical()
+    }
+
+    pub fn is_warning(&self) -> bool {
+        self.severity().is_warning()
+    }
+}
+
+impl Display for ErrorEntry {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ErrorEntry::Generic(err) => write!(f, "{}", err),
+            ErrorEntry::VoteAccount(err) => write!(f, "{}", err),
+        }
+    }
+}
+
+#[must_use = "ErrorHandlerBuilder must be consumed with .add()"]
+pub struct ErrorHandlerBuilder<'a> {
+    handler: &'a mut ErrorHandler,
+    severity: ErrorSeverity,
+    message: Option<String>,
+    source: Option<String>,
+    vote_account: Option<Pubkey>,
+}
+
+impl<'a> ErrorHandlerBuilder<'a> {
+    fn new(handler: &'a mut ErrorHandler, severity: ErrorSeverity) -> Self {
+        Self {
+            handler,
+            severity,
+            message: None,
+            source: None,
+            vote_account: None,
+        }
+    }
+
+    pub fn with_msg<D: Display>(mut self, message: D) -> Self {
+        self.message = Some(message.to_string());
+        self
+    }
+
+    pub fn with_err(mut self, err: anyhow::Error) -> Self {
+        self.message = Some(format!("{}", err));
+        self
+    }
+
+    pub fn with_source<S: Into<String>>(mut self, source: S) -> Self {
+        self.source = Some(source.into());
+        self
+    }
+
+    pub fn with_vote<V: Into<Pubkey>>(mut self, vote_account: V) -> Self {
+        self.vote_account = Some(vote_account.into());
+        self
+    }
+
+    pub fn add(self) {
+        let message = self
+            .message
+            .expect("Message is required - use with_msg() or with()");
```

Comment:
```suggestion
            .expect("Message is required - use with_msg() or with_err()");
```
No `with()` is present, I guess `with_err()` was meant.